lib/bolt/pal/yaml_plan.rb

```diff
@@ -36,23 +39,16 @@ class YamlPlan
           'required_keys' => Set.new
         }
       }.freeze
-
-      Parameter = Struct.new(:name, :value, :type_expr) do
-        def captures_rest
-          false
-        end
-      end
-
       attr_reader :name, :parameters, :steps, :return
 
       def initialize(name, plan)
         # Top-level plan keys aren't allowed to be Puppet code, so force them
         # all to strings.
-        plan = Bolt::Util.walk_keys(plan) { |key| stringify(key) }
+        plan = Bolt::Util.walk_keys(plan) { |key| self.class.stringify(key) }
```

Comment:
this is an anti-pattern since it won't be available on subclasses. It's also another hint that the transpiler should be more closely related to the yaml plan.